shapeshift/web
# feat: allow account selection for UTXO sell asset

## Description

- Adds account selection for UTXOs (BTC and LTC are currently enabled)
- Refactors `updateQuote` logic to use effect-based triggers of a closure instead of duplicate manual invocations
- Refactors periodic quote refresh logic to use a closure
- Fixes the `action` property of `TradeState` so that it actually reflects the state of the swapper

Note, some architecture changes were needed to make the swapper extensible enough to get this working.

## Notice


## Pull Request Type


## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/2040

## Risk

Refactors many things to do with quotes - we'll need a swapper regression test.

## Testing

1. Turn on the `Thor` feature flag
2. Select `BTC` or `LTC` as the sell asset whilst connected to a supported wallet (e.g. Native)
3. An account selection button should show, defaulting to this account with the highest balance
4. When performing a trade, the selected account should be used to trade from

## Screenshots (if applicable)

<img width="396" alt="Screen Shot 2022-08-18 at 6 25 25 pm" src="https://user-images.githubusercontent.com/97164662/185347671-44835cfe-e407-4f08-a4af-f809e4ded73b.png">

<img width="403" alt="Screen Shot 2022-08-18 at 6 26 56 pm" src="https://user-images.githubusercontent.com/97164662/185347965-ceaefb7c-54de-46bf-a1a1-62b2c73808e0.png">
